src/main/java/io/codelex/advancedtest/exercise1/CreditCard.java

```diff
@@ -0,0 +1,31 @@
+package io.codelex.advancedtest.exercise1;
+
+import java.math.BigDecimal;
+
+public class CreditCard extends Card{
+
+    private BigDecimal balance;
+
+    public CreditCard(int cardNumber, String owner, String ccvCode) {
+        super(cardNumber, owner, ccvCode);
+    }
+
+    @Override
+    public void addMoney(BigDecimal amount) {
+        balance = balance.add(amount);
+    }
+
+    @Override
+    public void withdrawMoney(BigDecimal amount) throws NotEnoughFundsException {
```

Comment:
Ar ko šī metode atšķiras no DebitCard withdrawMoney metodes? .. tikai ar to, ka šīs metodes beigās ir papildus pārbaude. Kā šo var uzlabot - to kopīgu daļu ieliec abstraktajā klasē kā parastu metodi (jebšu abstraktajā klasē jābūt metodei withdrawMoney, kura ir 1:1 tā kā Tu viņu esi uzrakstījis DebitCard klasē). Līdz ar to DebitCard var dzēst ārā to metodi, jo viņa jau būs no abstraktās klases, savukārt šeit CreditCard vari darīt šādi:
```
super.withdrawMoney(amount);
if(balance.compareTo(new BigDecimal("100")) < 0){
            System.out.println("Warning: Low funds");
        }
```
NB! Šis viss strādās tikai tad, ja balance mainīgo arī ieliksi abstraktajā klasē Card.